assert in julia.h raises warning

The commit 3f4be475 added some nice asserts, but gcc complains because

    julia.h: In function ‘jl_value_t* jl_tupleref(void*, size_t)’:
    julia.h:447:5: warning: comparison of unsigned expression >= 0 is always true [-Wtype-limits]
    julia.h: In function ‘jl_value_t* jl_tupleset(void*, size_t, void*)’:
    julia.h:452:5: warning: comparison of unsigned expression >= 0 is always true [-Wtype-limits]
    julia.h: In function ‘jl_value_t* jl_cellref(void*, size_t)’:
    julia.h:458:5: warning: comparison of unsigned expression >= 0 is always true [-Wtype-limits]
    julia.h: In function ‘jl_value_t* jl_cellset(void*, size_t, void*)’:
    julia.h:463:5: warning: comparison of unsigned expression >= 0 is always true [-Wtype-limits]

And it's true **size_t** is always unsigned [see here for instance](http://stackoverflow.com/questions/1089176/is-size-t-always-unsigned)

--- a/src/julia.h
+++ b/src/julia.h
@@ -444,23 +444,23 @@ DLLEXPORT size_t jl_array_len_(jl_array_t *a);
 
 STATIC_INLINE jl_value_t *jl_tupleref(void *t, size_t i)
 {
-    assert(i < jl_tuple_len(t) && i >= 0);
+    assert(i < jl_tuple_len(t));
     return jl_tuple_data(t)[i];
 }
 STATIC_INLINE jl_value_t *jl_tupleset(void *t, size_t i, void *x)
 {
-    assert(i < jl_tuple_len(t) && i >= 0);
+    assert(i < jl_tuple_len(t));
     jl_tuple_data(t)[i] = (jl_value_t*)x;
     return (jl_value_t*)x;
 }
 STATIC_INLINE jl_value_t *jl_cellref(void *a, size_t i)
 {
-    assert(i < jl_array_len(a) && i >= 0);
+    assert(i < jl_array_len(a));
     return ((jl_value_t**)(jl_array_data(a)))[i];
 }
 STATIC_INLINE jl_value_t *jl_cellset(void *a, size_t i, void *x)
 {
-    assert(i < jl_array_len(a) && i >= 0);
+    assert(i < jl_array_len(a));
     ((jl_value_t**)(jl_array_data(a)))[i] = (jl_value_t*)x;
     return (jl_value_t*)x;
 }

